src/sqs/SqsWorker.ts

```diff
@@ -0,0 +1,96 @@
+import * as SqsConsumer from 'sqs-consumer';
+import * as AWS from 'aws-sdk';
+import { PromiseUtil } from '../util/PromiseUtil';
+import { LogManager } from '../log/LogManager';
+import { Histogram, MetricsService } from '../metrics/Metrics';
+
+const log = LogManager.getLogger();
+
+export interface SqsConfigObject {
+  /**
+   * The queueUrl you are connecting to. (Default: localhost)
+   */
+  queueUrl?: string,
+
+  region?: string,
+
+  attributeNames?: Array<string> ,
+
+  handleMessage?: Function,
+
+  batchSize?: number,
+  /**
+   * The milliseconds before a timeout occurs during the initial connection to the MySQL server. (Default: 10 seconds)
+   */
+  connectTimeout?: number,
+}
+
+export abstract class SqsHandler {
+  /**
+   *
+   * @param message
+   * @param done
+   */
+  abstract handle(message: Object, done: (err?: Error) => void): void;
+}
+
+export class SqsWorker {
+
+  static startMethod = 'initialise';
+  static stopMethod = 'shutdown';
+  readonly maxRetries = 5;
+
+  configuration: SqsConfigObject;
+
+  name: string;
+
+  queueUrl: string;
+
+  consumerCreator: (config: SqsConfigObject) => SqsConsumer;
+
+  instance: SqsConsumer;
+
+  handler: SqsHandler;
+
+  constructor() {
+    this.name = 'NotSet';
+    this.consumerCreator = (config: SqsConfigObject) => SqsConsumer.create(config);
+  }
+
+  initialise() {
+    this.configuration = {
```

Comment:
How can you set the rest of the values in the configuration object? Why do it this way?
I'd suggest the constructor receives a SqsConfigObject, and you can easily have some default values by doing something like:
```
constructor(config: SqsConfigObject) {
  this.configuration = Object.assign(
    {}, 
    {attributeNames: ['All', 'ApproximateFirstReceiveTimestamp', 'ApproximateReceiveCount']}, 
    config);
}
```
this way any key that is set in the passed config will override what's there already and if not it'll use the default.